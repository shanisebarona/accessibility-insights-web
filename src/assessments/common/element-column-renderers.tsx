// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { AssessmentInstanceDetailsColumn } from 'DetailsView/components/assessment-instance-details-column';
import * as React from 'react';

export function onRenderPathColumn(item: InstanceTableRow): JSX.Element {
    let textContent = '';
    if (item.instance.target) {
        textContent = item.instance.target.join(';');
    }

    return (
        <AssessmentInstanceDetailsColumn
            textContent={textContent}
            customClassName="not-applicable"
        />
    );
}

export function onRenderSnippetColumn(item: InstanceTableRow): JSX.Element {
    return (
        <AssessmentInstanceDetailsColumn
            textContent={item.instance.html}
            customClassName="not-applicable"
        />
    );
}
