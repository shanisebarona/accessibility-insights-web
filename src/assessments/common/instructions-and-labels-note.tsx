// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import * as Markup from '../markup';

export const InstructionsAndLabelsNotes = () => (
    <Markup.Emphasis>
        Note: Both WCAG 2.0 and 2.1 require a widget's visible label and instructions (if present)
        to be programmatically determinable. WCAG 2.1 also requires a widget's visible label and
        instructions (if present) to be included in its accessible name and description.
    </Markup.Emphasis>
);
