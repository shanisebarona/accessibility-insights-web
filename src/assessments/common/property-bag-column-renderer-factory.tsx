// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { propertyBagColumnRenderer } from './property-bag-column-renderer';

export class PropertyBagColumnRendererFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: InstanceTableRow<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}
