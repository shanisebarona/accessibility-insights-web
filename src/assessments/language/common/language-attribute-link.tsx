// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NewTabLink } from 'common/components/new-tab-link';
import * as React from 'react';

export const LanguageAttributeLink = () => (
    <NewTabLink href="http://www.iana.org/assignments/language-subtag-registry/language-subtag-registry">
        language attribute
    </NewTabLink>
);
