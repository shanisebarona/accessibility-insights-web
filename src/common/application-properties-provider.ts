// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ToolData } from './types/store-data/unified-data-interface';

export const createToolData = (
    scanEngineName: string,
    scanEngineVersion: string,
    toolName: string,
    toolVersion?: string,
    environmentName?: string,
    resolution?: string,
): ToolData => {
    return {
        scanEngineProperties: {
            name: scanEngineName,
            version: scanEngineVersion,
        },
        applicationProperties: {
            name: toolName,
            version: toolVersion,
            environmentName: environmentName,
            resolution: resolution,
        },
    };
};
