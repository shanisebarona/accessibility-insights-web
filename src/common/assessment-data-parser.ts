// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';

export class AssessmentDataParser {
    public parseAssessmentData = (assessmentData: string): VersionedAssessmentData => {
        const loadedAssessmentData: VersionedAssessmentData = JSON.parse(assessmentData);
        return loadedAssessmentData;
    };
}
