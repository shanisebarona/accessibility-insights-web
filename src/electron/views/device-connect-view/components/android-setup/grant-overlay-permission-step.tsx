// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { AndroidSetupSpinnerStep } from 'electron/views/device-connect-view/components/android-setup/android-setup-spinner-step';
import * as React from 'react';
import { CommonAndroidSetupStepProps } from './android-setup-types';

export const GrantOverlayPermissionStep = NamedFC<CommonAndroidSetupStepProps>(
    'GrantOverlayPermissionStep',
    (props: CommonAndroidSetupStepProps) => {
        return (
            <AndroidSetupSpinnerStep
                deps={props.deps}
                spinnerLabel="Granting overlay permission..."
                disableCancel={true}
            />
        );
    },
);
