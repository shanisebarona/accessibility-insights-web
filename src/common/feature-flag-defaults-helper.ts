// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { FeatureFlagDetail } from 'common/feature-flags';
import { forEach } from 'lodash';

import { FeatureFlagStoreData } from './types/store-data/feature-flag-store-data';

export class FeatureFlagDefaultsHelper {
    public constructor(private readonly getFeatureFlagDetails: () => FeatureFlagDetail[]) {}

    public getDefaultFeatureFlagValues(): FeatureFlagStoreData {
        const details: FeatureFlagDetail[] = this.getFeatureFlagDetails();
        const values: FeatureFlagStoreData = {};
        forEach(details, detail => {
            values[detail.id] = detail.defaultValue;
        });
        return values;
    }

    public getForceDefaultFlags(): string[] {
        const details: FeatureFlagDetail[] = this.getFeatureFlagDetails();
        const forceDefaultFlags: string[] = [];
        forEach(details, detail => {
            if (detail.forceDefault) {
                forceDefaultFlags.push(detail.id);
            }
        });
        return forceDefaultFlags;
    }
}
