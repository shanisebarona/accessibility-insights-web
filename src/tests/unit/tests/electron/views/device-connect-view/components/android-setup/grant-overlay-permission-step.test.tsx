// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { GrantOverlayPermissionStep } from 'electron/views/device-connect-view/components/android-setup/grant-overlay-permission-step';
import { shallow } from 'enzyme';
import * as React from 'react';
import { AndroidSetupStepPropsBuilder } from 'tests/unit/common/android-setup-step-props-builder';

describe('GrantOverlayPermissionStep', () => {
    const props = new AndroidSetupStepPropsBuilder('grant-overlay-permission').build();

    it('renders', () => {
        const rendered = shallow(<GrantOverlayPermissionStep {...props} />);
        expect(rendered.getElement()).toMatchSnapshot();
    });
});
