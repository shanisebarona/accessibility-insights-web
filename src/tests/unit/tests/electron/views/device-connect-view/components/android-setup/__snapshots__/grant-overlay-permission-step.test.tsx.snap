// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrantOverlayPermissionStep renders 1`] = `
<AndroidSetupSpinnerStep
  deps={
    Object {
      "LinkComponent": [Function],
    }
  }
  disableCancel={true}
  spinnerLabel="Granting overlay permission..."
/>
`;
